### PR TITLE
Add support for multiple Etags in If-None-Match header

### PR DESCRIPTION
**Pull request by straight-shoota**
This enhances Etag supported (which as added in #6145) to recognize multiple values in the `If-None-Match` header according to [RFC 2732](https://tools.ietf.org/html/rfc7232).

The header field parser is reusable outside `StaticFileHandler` as `Headers#if_none_match` and `Headers#if_match`.

### CODE REVIEW

**Comment by sdogruyol** [comment]
LGTM :+1: I especially liked how it's not coupled to `StaticFileHandler` :100:

**Comment by straight-shoota** [comment]
@sdogruyol Yeah, all this logic for handling cached requests should eventually be moved to `Request`/`Response`. Any handler can just set `Last-Modified` and/or `Etag` on the response and then check the request headers against that. The only issue is that this check needs access to both request and response, so it probably needs to go on `HTTP::Server::Context`.

File: src/http/headers.cr [hunk 1]
```diff
@@ -258,6 +258,73 @@ struct HTTP::Headers
 
   forward_missing_to @hash
 
+  def if_none_match : Array(String)?
```
Comment (RX14) [inline src/http/headers.cr]: I don't think this parsing method belongs here.

Comment (straight-shoota) [reply src/http/headers.cr]: Yeah, I wasn't sure whether this should go here or in `Request`. I'll move it to the latter.

Comment (RX14) [reply src/http/headers.cr]: This wasn't removed.

**Comment by straight-shoota** [comment]
Updated

**Comment by straight-shoota** [comment]
Merge?

**Comment by RX14** [comment]
need another review and bcardiff is on holiday for the next 3 weeks.

**Comment by RX14** [comment]
Because the PR has had extra commits since then...

**Comment by asterite** [comment]
Approved :-P

(someone probably needs to add me as a core team member again, every two or three months I get tired of the language and remove myself, though I'm trying to manage it better now)

**Comment by straight-shoota** [comment]
Oh, yes, that's true. My bad. @sdogruyol care for another look?

**Comment by sdogruyol** [comment]
LGTM :+1:  Thank you @straight-shoota :pray:

**Pull request merged**